presto-main/src/main/java/com/facebook/presto/server/protocol/Query.java

```diff
@@ -408,10 +414,15 @@ else if (queryManager.getQueryRetryCount(queryId) == 1 && retryQueryWithHistoryB
 
             if (previousQueryTopLevelPlanHash.isPresent() && currentTopLevelPlanHash.isPresent() && currentTopLevelPlanHash.equals(previousQueryTopLevelPlanHash)
                     || (!previousQueryTopLevelPlanHash.isPresent() && !currentTopLevelPlanHash.isPresent())) {
-                queryManager.failQuery(queryId, new PrestoException(GENERIC_INTERNAL_ERROR, "Since the plan hashes did not change, your retry query will not execute." +
+                String retryQuerySamePlanMessage = "HBO did not find better plan aborting retry";
+                session.getWarningCollector().add(new PrestoWarning(CURRENT_QUERY_IS_RETRY_QUERY, retryQuerySamePlanMessage));
+
+                queryManager.failQuery(queryId, new PrestoException(GENERIC_INTERNAL_ERROR, "HBO did not find better plan aborting retry." +
```

Comment:
Don't make this internal error - maybe just make it a info/logging if the user doesn't see it - which they shouldn't

Reply:
@kaikalur https://github.com/prestodb/presto/pull/23372

Reply:
Not sure what that issue says but in general we shouldnt use internal error as it's counted as something went wrong. Here nothing went wrong. So I suggest just info logging it. User doesn't need to know